tests/unit/plugins/modules/net_tools/pritunl/test_pritunl_user.py

```diff
@@ -0,0 +1,208 @@
+    AnsibleFailJson,
+    ModuleTestCase,
+    set_module_args,
+)
+
+__metaclass__ = type
+
+
+def mock_pritunl_api(func, **kwargs):
+    def wrapped(self=None):
+        with self.patch_get_pritunl_organizations(
+            side_effect=PritunlListOrganizationMock
+        ):
+            with self.patch_get_pritunl_users(side_effect=PritunlListUserMock):
+                with self.patch_add_pritunl_users(side_effect=PritunlPostUserMock):
+                    with self.patch_delete_pritunl_users(
+                        side_effect=PritunlDeleteUserMock
+                    ):
+                        func(self, **kwargs)
+
+    return wrapped
+
+
+class TestPritunlUser(ModuleTestCase):
+    def setUp(self):
+        super(TestPritunlUser, self).setUp()
+        self.module = pritunl_user
+
+        # Add backward compatibility
+        if sys.version_info < (3, 2):
+            self.assertRegex = self.assertRegexpMatches
+
+    def tearDown(self):
+        super(TestPritunlUser, self).tearDown()
+
+    def patch_get_pritunl_users(self, **kwds):
+        return patch(
+            "ansible_collections.community.general.plugins.module_utils.net_tools.pritunl.api._get_pritunl_users",
+            autospec=True,
+            **kwds
+        )
+
+    def patch_add_pritunl_users(self, **kwds):
+        return patch(
+            "ansible_collections.community.general.plugins.module_utils.net_tools.pritunl.api._post_pritunl_user",
+            autospec=True,
+            **kwds
+        )
+
+    def patch_update_pritunl_users(self, **kwds):
+        return patch(
+            "ansible_collections.community.general.plugins.module_utils.net_tools.pritunl.api._put_pritunl_user",
+            autospec=True,
+            **kwds
+        )
+
+    def patch_delete_pritunl_users(self, **kwds):
+        return patch(
+            "ansible_collections.community.general.plugins.module_utils.net_tools.pritunl.api._delete_pritunl_user",
+            autospec=True,
+            **kwds
+        )
+
+    def patch_get_pritunl_organizations(self, **kwds):
+        return patch(
+            "ansible_collections.community.general.plugins.module_utils.net_tools.pritunl.api._get_pritunl_organizations",
+            autospec=True,
+            **kwds
+        )
+
+    def test_without_parameters(self):
+        """Test without parameters"""
+        set_module_args({})
+        with self.assertRaises(AnsibleFailJson):
+            self.module.main()
+
+    @mock_pritunl_api
+    def test_present(self):
+        """Test Pritunl user creation and update."""
+        user_params = {
+            "user_name": "alice",
+            "user_email": "alice@company.com",
+        }
+        set_module_args(
+            dict_merge(
+                {
+                    "pritunl_api_token": "token",
+                    "pritunl_api_secret": "secret",
+                    "pritunl_url": "https://pritunl.domain.com",
+                    "organization": "GumGum",
```

Comment:
Night at the Museum? :rofl:

Reply:
Nop my company name -> https://gumgum.com/ 😛

Reply:
Ha, once we have kids and start watching those movies, we see them everywhere :-P